src/pages/addUpdateBook/types/interface.ts

```diff
@@ -0,0 +1,84 @@
+import { SwapType } from "./enum";
+
+export type TOptions = {
+  label: string;
+  value: string;
+};
+export interface IBookDetailsProps {
+  languageOptions: TOptions[] | undefined;
+  conditionOptions: TOptions[] | undefined;
+}
+export interface IBook {
+  bookTitle: string;
+  authorName: string;
+  byBookCover: string | File;
+}
+export interface IAddUpdateBookData {
+  books: IBook[];
+  favGenres: string[];
+  conditionType: SwapType;
```

Comment:
change conditionType to swapType everywhere

```suggestion
  swapType: SwapType;
```